samples/initialize_server.py

```diff
@@ -0,0 +1,105 @@
+####
+# This script sets up a server. It uploads datasources and workbooks from the local filesystem.
+#
+# By default, all content is published to the Default project on the Default site.
+####
+
+import tableauserverclient as TSC
+import argparse
+import getpass
+import logging
+import glob
+
+def main():
+    parser = argparse.ArgumentParser(description='Initialize a server with content.')
+    parser.add_argument('--server', '-s', required=True, help='server address')
+    parser.add_argument('--datasources-folder', '-df', required=True, help='folder containing datasources')
+    parser.add_argument('--workbooks-folder', '-wf', required=True, help='folder containing workbooks')
+    parser.add_argument('--site', '-si', required=False, default='Default', help='site to use')
+    parser.add_argument('--project', '-p', required=False, default='Default', help='project to use')
+    parser.add_argument('--username', '-u', required=True, help='username to sign into server')
+    parser.add_argument('--logging-level', '-l', choices=['debug', 'info', 'error'], default='error',
+                        help='desired logging level (set to error by default)')
+    args = parser.parse_args()
+
+    password = getpass.getpass("Password: ")
+
+    # Set logging level based on user input, or error by default
+    logging_level = getattr(logging, args.logging_level.upper())
+    logging.basicConfig(level=logging_level)
+
+    ################################################################################
+    # Step 1: Sign in to server.
+    ################################################################################
+    tableau_auth = TSC.TableauAuth(args.username, password)
+    server = TSC.Server(args.server)
+
+    with server.auth.sign_in(tableau_auth):
+
+        ################################################################################
+        # Step 2: Create the site we need only if it doesn't exist
+        ################################################################################
+        print("Checking to see if we need to create the site...")
+
+        all_sites, _ = server.sites.get()
+        existing_site = next((s for s in all_sites if s.name == args.site), None)
+
+        # Create the site if it doesn't exist
+        if existing_site is None:
+            print("Site not found: {0} Creating it...").format(args.site)
+            new_site = TSC.SiteItem(name=args.site, content_url=args.site.replace(" ", ""), admin_mode=TSC.SiteItem.AdminMode.ContentAndUsers)
```

Comment:
It's better practice to encourage providing a site_id, explicitly, I think, instead of guessing from the name. That's what tabcmd does and it makes it easier to debug the character issues you ran into